server/service/queries_test.go

```diff
@@ -114,6 +115,174 @@ func TestListQueries(t *testing.T) {
 	}
 }
 
+// used in ModifyQuery and NewQuery
+func TestQueryPayloadValidation(t *testing.T) {
+	ds := new(mock.Store)
+	svc, ctx := newTestService(t, ds, nil, nil)
+
+	testCases := []struct {
+		name         string
+		queryPayload fleet.QueryPayload
+		shouldErr    bool
+	}{
+		{
+			"All valid",
+			fleet.QueryPayload{
+				Name:     ptr.String("test query"),
+				Query:    ptr.String("select 1"),
+				Logging:  ptr.String("snapshot"),
+				Platform: ptr.String(""),
+			},
+			false,
+		},
+		{
+			"Invalid  - empty string name",
+			fleet.QueryPayload{
+				Name:     ptr.String(""),
+				Query:    ptr.String("select 1"),
+				Logging:  ptr.String("snapshot"),
+				Platform: ptr.String(""),
+			},
+			true,
+		},
+		// {
+		// 	"Invalid SQL",
+		// 	{
+		// 		// TODO
+		// 	},
+		// 	true,
+		// },
+		// {
+		// 	"Invalid logging",
+		// 	{
+		// 		// TODO
+		// 	},
+		// 	true,
+		// },
+		// {
+		// 	"Invalid platform 0",
+		// 	{
+		// 		// TODO
+		// 	},
+		// 	true,
+		// },
+		// {
+		// 	"Invalid platform 1",
+		// 	{
+		// 		// TODO
+		// 	},
+		// 	true,
+		// },
+		// {
+		// 	"Invalid platform 2",
+		// 	{
+		// 		// TODO
+		// 	},
+		// 	true,
+		// },
+		// {
+		// 	"Invalid platform 3",
+		// 	{
+		// 		// TODO
+		// 	},
+		// 	true,
+		// },
+	}
+
+	// testAdmin := fleet.User{
+	// 	ID:    1,
+	// 	Teams: []fleet.UserTeam{},
+	// }
+
+	fmt.Printf("running tests\n----\n")
+	for _, tt := range testCases {
+		t.Run(tt.name, func(t *testing.T) {
+			// viewerCtx := viewer.NewContext(ctx, viewer.Viewer{User: &testAdmin})
+			fmt.Printf("running test: %v\n----\n", &tt)
```

Comment:
No need to take the address of `tt` (`&tt`), just `tt` here.

Reply:
@lucasmrod I'm having trouble getting these tests to work. Am I headed in the right direction? Is the integration test at `server/service/integration_core_test.go/TestQueriesBadRequests` sufficient in place of this?